### PR TITLE
Update engines.yml

### PR DESCRIPTION
**Pull request by cefoo**
# Description
## Type of PR
- Edits _engines.yml_

### Checklist:

- [x] I have read the [contributing guidelines](contributing.md).
- [x] I have followed the [style guide](http://machinetranslate.org/style).
- [x] I have made sure that the URL is not already in use.
- [x] I have cross-linked relevant articles.
- [x] I have used the UK spelling.
- [x] I have kept consistency with the structure of sister articles in the same directory.

### CODE REVIEW

**Review by bittlingmayer — APPROVED** [review]
A fixed something in YAML, next time you should use a validator and eventually we should add an automatic check.

I would say:

- We could change `domain` to `url` (and prepend `https://` to them), because many of them are URLs i.e. pages, not just domains i.e. sites.

- The source URL is actually useful, no need to comment it out.  We could call it `language_support_url`?  Or `info_url`?  (Not sure which it is.).  And just leave it empty or use the email URL (with `mailto:`) for those that were confirmed via email.

**Comment by bittlingmayer** [comment]
Very excited about this one!

File: _data/engines.yml [hunk 1]
```diff
@@ -1,4 +1,236 @@
...
+  info_url: https://www.alibabacloud.com/help/en/machine-translation/latest/general-version-of-machine-translation
+
+- name: Amazon Translate
+  url: http://aws.amazon.com/es/translate
+  languages: [ af, am, ar, az, bg, bn, bs, ca, cs, cy, da, de, el, en, es, es-mx, et, fa, fa-af, fi, fr, fr-ca, ga, gu, ha, he, hi, hr, ht, hu, hy, id, is, it, ja, ka, kk, kn, ko, lt, lv, mk, ml, mn, mr, ms, mt, nl, no, pa, pl, ps, pt, pt-pt, ro, ru, si, sk, sl, so, sq, sr, sv, sw, ta, te, th, tl, tr, uk, ur, uz, vi, zh, zh-tw ]
+  info_url: https://docs.aws.amazon.com/translate/latest/dg/what-is-languages.html
+
+- name: Apertium
+  url: http://www.apertium.org
+  languages: [ afr-nld, arg-cat, bel-rus, br-fr, cat-ita, cat-srd, crh-tur, cym-eng, dan-nor, eng-cat, eng-kaz, en-gl, eng-spa, eo-ca, eo-en, eo-es, eo-fr, es-gl, es-pt, es-ro, eu-en, eu-es, fra-cat, fr-es, hbs-eng, hbs-mkd, hbs-slv, ind-zlm, isl-eng, isl-swe, kaz-rus, kaz-tat, mkd-bul, mkd-eng, mlt-ara, nno-nob, oc-ca, oc-es, oci-fra, pol-szl, por-cat, pt-gl, ron-cat, rus-ukr, sme-nob, spa-arg, spa-ast, spa-cat, spa-ita, srd-ita, swe-dan, swe-nor, urd-hin ]
```
Comment (bittlingmayer) [inline _data/engines.yml:22]: Need to fix this

```
[ [ af, nl ] ...
```

File: _data/engines.yml [hunk 2]
```diff
@@ -1,4 +1,236 @@
...
+
+- name: Youdao Translate
+  url: http://fanyi.youdao.com
+  languages: [ af, am, ar, az, be, bg, bn, bs, ca, ceb, co, cs, cy, da, de, el, en, eo, es, et, eu, fa, fi, fj, fr, fy, ga, gd, gl, gu, ha, haw, he, hi, hr, ht, hu, hy, id, ig, is, it, ja, jw, ka, kk, km, kn, ko, ku, ky, la, lb, lo, lt, lv, mg, mi, mk, ml, mn, mr, ms, mt, mww, my, ne, nl, no, ny, otq, pa, pl, ps, pt, ro, ru, sd, si, sk, sl, sm, sn, so, sq, sr-Cyrl, sr-Latn, st, su, sv, sw, ta, te, tg, th, tl, tlh, to, tr, ty, uk, ur, uz, vi, xh, yi, yo, yua, yue, zh-cn, zh-tw, zu ]
+  info_url: https://ai.youdao.com/DOCSIRMA/html/%E8%87%AA%E7%84%B6%E8%AF%AD%E8%A8%80%E7%BF%BB%E8%AF%91/API%E6%96%87%E6%A1%A3/%E6%96%87%E6%9C%AC%E7%BF%BB%E8%AF%91%E6%9C%8D%E5%8A%A1/%E6%96%87%E6%9C%AC%E7%BF%BB%E8%AF%91%E6%9C%8D%E5%8A%A1-API%E6%96%87%E6%A1%A3.html
+
+- name: Wordlingo
+  url: https://www.worldlingo.com/en/products/worldlingo_api.html
+  languages: [ af, ar, az, be, bg, bn, ca, cs, cy, da, de, el, en, eo, es, et, eu, fa, fi, fr, ga, gl, gu, ha, he, hi, hr, ht, hu, id, is, it, ja, ka, kn, ko, la, lt, lv, mk, ms, mt, nl, no, pl, pt, ro, ru, sk, sl, so, sq, sr, sv, sw, ta, te, th, tl, tr, uk, ur, vi, yi, zh_cn, zh_tw ]
```
Comment (bittlingmayer) [inline _data/engines.yml:235]: Should be `-` not `_`, but I guess we could make the other script deal with that.

File: _data/engines.yml [hunk 3]
```diff
@@ -1,4 +1,236 @@
...
+  info_url: https://cloud.tencent.com/document/api/551/15619
+
+- name: TextShuttle
+  url: https://www.textshuttle.ai/
+  languages: [ de, en, es, fr, it ]
```
Comment (bittlingmayer) [inline _data/engines.yml:186]: This is difficult.  They offer "any" other desired languages.

File: _data/engines.yml [hunk 4]
```diff
@@ -1,4 +1,236 @@
 ---
+- name: AISA
+  url: http://www.aisa.digital/machine-translation-api
+  languages: [ ar, de, es-es, fr-fr, hi, huyu, id, it, ja, ko, ms, pt-pt, ru, th, vi, zh-cn, zh-tw ]
```
[the remark this answers is not in the document]
Comment (cefoo) [reply _data/engines.yml:5]: Yes, it's ISO 636-6 for Chinese from Singapore, which AISA supports.
[wikipedia.org/wiki/Singaporean_Mandarin](https://en.wikipedia.org/wiki/Singaporean_Mandarin)

File: _data/engines.yml (hunk 1 above)
Comment (cefoo) [reply _data/engines.yml]: Fixed!
I will add 3-letter codes to languages.yml too, especially for Niutrans supported languages.

File: _data/engines.yml [hunk 5]
```diff
@@ -1,4 +1,236 @@
...
+  languages: [ afr-nld, arg-cat, bel-rus, br-fr, cat-ita, cat-srd, crh-tur, cym-eng, dan-nor, eng-cat, eng-kaz, en-gl, eng-spa, eo-ca, eo-en, eo-es, eo-fr, es-gl, es-pt, es-ro, eu-en, eu-es, fra-cat, fr-es, hbs-eng, hbs-mkd, hbs-slv, ind-zlm, isl-eng, isl-swe, kaz-rus, kaz-tat, mkd-bul, mkd-eng, mlt-ara, nno-nob, oc-ca, oc-es, oci-fra, pol-szl, por-cat, pt-gl, ron-cat, rus-ukr, sme-nob, spa-arg, spa-ast, spa-cat, spa-ita, srd-ita, swe-dan, swe-nor, urd-hin ]
+  info_url: https://wiki.apertium.org/wiki/List_of_language_pairs
+
+- name: AppTek
+  url: http://www.apptek.com
+  languages: [ af, am, ar, az, be, bg, bn, bs, cs, da, de, el, en, es-es, es-mx, es-us, et, fa, fa-prs, fi, fr, fr-ca, fr-fr, gu, ha, he, hi, hr, hu, hy, id, is, it, ja, ka, kk, kn, ko, ky, lt, lv, mk, ml, mn, mr, ms, nl, no, pa, pl, ps, pt-br, pt-pt, ro, ru, sk, sl, so, sq, sr, sr-Latn, sv, te, th, ti, tk, tl, tr, uk, ur, uz, zh-cn, zh-tw ]
+  info_url: https://www.apptek.com/technology/apptek-languages
+
+- name: Baidu
+  url: http://fanyi.baidu.com
```
[the remark this answers is not in the document]
Comment (cefoo) [reply _data/engines.yml]: You're right, it's fixed now.

File: _data/engines.yml [hunk 6]
```diff
@@ -1,4 +1,236 @@
...
+  info_url: https://docs.lingvanex.com/reference/user-guide
+
+- name: Microsoft Translator
+  url: http://translator.microsoft.com
+  languages: [ af, am, ar, as, az, ba, bg, bn, bo, bs, ca, cs, cy, da, de, dv, el, en, es, et, eu, fa, fi, fil, fj, fo, fr, fr-ca, ga, gl, gu, he, hi, hr, hsb, ht, hu, hy, id, ikt, is, it, iu, iu-Latn, ja, ka, kk, km, kmr, kn, ko, ku, ky, lo, lt, lv, lzh, mg, mi, mk, ml, mn-Cyrl, mn-Mong, mr, ms, mt, mww, my, nb, ne, nl, or, otq, pa, pl, prs, ps, pt, pt-pt, ro, ru, sk, sl, sm, so, sq, sr-Cyrl, sr-Latn, sv, sw, ta, te, th, ti, tk, tlh-Latn, tlh-Piqd, to, tr, tt, ty, ug, uk, ur, uz, vi, yua, yue, zh-Hans, zh-Hant, zu ]
+  info_url: https://docs.microsoft.com/en-us/azure/cognitive-services/translator/language-support
+
+- name: Microsoft Translator (custom)
+  url: http://translator.microsoft.com
+  languages: [ af, ar, as, bg, bn, bs, ca, cs, cy, da, de, el, en, es, et, fa, fi, fil, fj, fr, ga, gu, he, hi, hr, hu, id, is, it, iu, ja, kn, ko, lt, lv, mg, mi, ml, mr, ms, mt, nb, nl, or, pa, pl, pt, ro, ru, sk, sl, sm, sr-Latn, sv, sw, ta, te, th, to, tr, ty, uk, ur, vi, zh-Hans, zh-Hant ]
```
[the remark this answers is not in the document]
Comment (cefoo) [reply _data/engines.yml]: Sure, done.

File: _data/engines.yml (hunk 3 above)
Comment (cefoo) [reply _data/engines.yml:279]: Right, I considered stock engines for this page, as many engines, like KantanMT, offer customizable languages as per their client's request.
I added `custom_languages: null`.

File: _data/engines.yml (hunk 2 above)
Comment (cefoo) [reply _data/engines.yml:343]: Fixed

File: _data/engines.yml [hunk 7]
```diff
@@ -1,4 +1,345 @@
...
+  url: https://papago.naver.com/business
+  languages: [ [ko, en], [ko, zh-cn], [ko, zh-tw], [ko, fr], [ko, es], [ko, vi], [ko, th], [ko, id], [ko, ja], [ko, ru], [ko, de], [ko, it], [en, ja], [en, fr], [zh, en], [zh, ja] ]
+  glossary: true
+  formality: true
+  self-serve: true
+  info_url: https://www.ncloud.com/product/aiService/papagoTranslation
+
+- name: Promt
+  id: promt
+  url: http://www.promt.com
```
Comment (bittlingmayer) [inline _data/engines.yml:226]: Let's us HTTPS unless it's absolutely not supported

File: _data/engines.yml (hunk 3 above)
Comment (bittlingmayer) [reply _data/engines.yml:279]: Maybe we should put `*`?

**Review by bittlingmayer — APPROVED** [review]

**Pull request merged**